src/smtlib_reader.cpp

```diff
@@ -39,12 +39,30 @@ const unordered_map<string, vector<string>> logic_map(
       { "RDL", { "RA" } },
       { "UF", { "UF" } } });
 
+// maps logic string to vector of associated SortKinds for that logic
+const unordered_map<string, vector<SortKind>> logic_sortkind_map(
+    { { "A", { ARRAY } },
+      { "AX", { ARRAY } },
+      { "BV", { BV } },
+      { "IDL", { INT } },
+      { "LIA", { INT } },
+      { "LIRA", { INT, REAL } },
+      { "LRA", { REAL } },
+      { "NIA", { INT } },
+      { "NIRA", { INT, REAL } },
+      { "NRA", { REAL } },
+      { "RDL", { REAL } },
+      { "UF", { FUNCTION } } });
+
 SmtLibReader::SmtLibReader(smt::SmtSolver & solver, bool strict)
     : solver_(solver),
       strict_(strict),
+      logic_("UNSET"),
```

Comment:
Maybe document above the `logic_` field that there is a special `"UNSET"` value.

Reply:
Good idea! Done in 03331c979f6c972fe0924e6c471856f332ce1cab.